util-hadoop/src/main/java/com/google/cloud/hadoop/util/HadoopConfigurationProperty.java

```diff
@@ -80,6 +92,14 @@ public String getPassword(Configuration config) {
     return logProperty(lookupKey, value);
   }
 
+  public Map<String, String> getHttpRequestHeaders(Configuration config) {
```

Comment:
It's kind of tricky situation here bcuz either way a new method is required. I added a getAll instead of get bcuz getPropsWithPrefix needs Function instead of BiFunction and I cannot reuse get method name for it either cuz config::get will confuse which get the code is trying to invoke.

Reply:
I end up with adding `getPropsWithPrefix` to the property class, it seems a little more consistent.